packages/color-slider/src/ColorSlider.ts

```diff
@@ -0,0 +1,345 @@
+            //this.value = parseFloat((color as HSV).h.toString());
+        } else {
+            const { h } = this._color.toHsv();
+            this.value = h;
+        }
+        this._previousColor = oldValue;
+        this.requestUpdate('color', oldValue);
+    }
+
+    private _color = new TinyColor({ h: 0, s: 1, v: 1 });
+
+    private _previousColor = new TinyColor({ h: 0, s: 1, v: 1 });
+
+    private _format: { format: string; isString: boolean } = {
+        format: '',
+        isString: false,
+    };
+
+    @property({ type: Number })
+    public step = 1;
+
+    private get altered(): number {
+        return this._altered;
+    }
+
+    private set altered(altered: number) {
+        this._altered = altered;
+        this.step = Math.max(1, this.altered * 10);
+    }
+
+    private _altered = 0;
+
+    private altKeys = new Set();
+
+    @query('input')
+    public input!: HTMLInputElement;
+
+    public get focusElement(): HTMLInputElement {
+        return this.input;
+    }
+
+    private handleKeydown(event: KeyboardEvent): void {
+        event.preventDefault();
+        const { key } = event;
+        if (['Shift', 'Meta', 'Control', 'Alt'].includes(key)) {
+            this.altKeys.add(key);
+            this.altered = this.altKeys.size;
+        }
+        let delta = 0;
+        switch (key) {
+            case 'ArrowUp':
+                delta = this.step;
+                break;
+            case 'ArrowDown':
+                delta = -this.step;
+                break;
+            case 'ArrowLeft':
+                delta = this.step * (this.isLTR ? -1 : 1);
+                break;
+            case 'ArrowRight':
+                delta = this.step * (this.isLTR ? 1 : -1);
+                break;
+        }
+        this.sliderHandlePosition = Math.min(
+            100,
+            Math.max(0, this.sliderHandlePosition + delta)
+        );
+    }
+
+    private handleKeyup(event: KeyboardEvent): void {
+        event.preventDefault();
+        const { key } = event;
+        if (['Shift', 'Meta', 'Control', 'Alt'].includes(key)) {
+            this.altKeys.delete(key);
+            this.altered = this.altKeys.size;
+        }
+    }
+
+    private handleFocus(): void {
+        this.focused = true;
+    }
+
+    private handleBlur(): void {
+        this.focused = false;
+    }
+
+    private boundingClientRect!: DOMRect;
+
+    private handlePointerdown(event: PointerEvent): void {
+        this._previousColor = this._color.clone();
+        this.boundingClientRect = this.getBoundingClientRect();
+        (event.target as HTMLElement).setPointerCapture(event.pointerId);
+    }
+
+    private handlePointermove(event: PointerEvent): void {
+        this.sliderHandlePosition = this.calculateHandlePosition(event);
+        this.value = 360 * (this.sliderHandlePosition / 100);
+
+        //this.color = `hsl(${this.value}, 100%, 50%)`;
+        this._color = new TinyColor({ h: this.value, s: '100%', l: '50%' });
```

Comment:
This should likely take `s` and `l` from the previous value of `this._color` so that we don't alter those parts of the `color` value?